Patch to work on 10.10 and earlier (and only those systems).
The undef prevents Metal from being imported; the defines fill in for
NSApplication.h, which otherwise has a conflicting type with IOSurfaceRef.

See: https://trac.macports.org/ticket/67263

--- lib/irrlicht/source/Irrlicht/MoltenVK.mm.orig
+++ lib/irrlicht/source/Irrlicht/MoltenVK.mm
@@ -1,9 +1,11 @@
 #include "MoltenVK.h"
 #include "SDL_vulkan.h"
+#undef __OBJC__
 #include "vulkan_wrapper.h"
 
 #include <dlfcn.h>
-#import <AppKit/NSApplication.h>
+extern const double NSAppKitVersionNumber;
+#define NSAppKitVersionNumber10_10_Max 1349
 
 #ifdef DLOPEN_MOLTENVK
 PFN_vkGetMoltenVKConfigurationMVK vkGetMoltenVKConfigurationMVK = NULL;
